helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -602,4 +609,53 @@ public static String getSchemaFullName(Schema schema) {
       return schema.getName();
     }
   }
+
+  /**
+   * Create a new schema field with provided arguments. This adds compatibility layer for {@link Schema.Field} before
+   * avro 1.9, which takes JsonNode as default value in constructor instead of Object.
+   * @param name the name of the field
+   * @param schema the schema of the field
+   * @param doc the doc of the field
+   * @param defaultValue the default value of the field. For avro 1.4-1.8, user should pass in a JsonNode. For avro 1.9
+   * and newer, user should pass in the default value object directly.
+   * @param order the order of the field.
+   * @return fully constructed Field instance
+   */
+  public static Schema.Field createSchemaField(String name, Schema schema, String doc, Object defaultValue, Schema.Field.Order order) {
+    FieldBuilder fieldBuilder;
+    switch (DETECTED_VERSION) {
```

Comment:
I think this kind of switching logic belongs inside each adapter.

Reply:
+1. the general pattern here is delegate all calls to the correct adapter.